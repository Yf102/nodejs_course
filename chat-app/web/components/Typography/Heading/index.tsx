import React from 'react'
import { twMerge } from 'tailwind-merge'

import styles from './Heading.module.scss'

type HeadingTag = 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6'
export type HeadingProps = {
  id?: string
  className?: string
  level: 1 | 2 | 3 | 4 | 5 | 6
  fontWeight?: 'normal' | 'bold'
  children?: React.ReactNode
  extraAttributes?: { [key: string]: string | number | boolean }
  style?: React.CSSProperties
}

/**
 * Creates a heading tag (`<h1>`, `<h2>`, etc.).
 */
const Heading = ({
  id,
  className,
  level = 1,
  children,
  style,
  ...extraAttributes
}: HeadingProps) => {
  const H = `h${level}` as HeadingTag

  const sizeMap = {
    1: 'text-3xl sm:text-4xl mb-5 font-semibold uppercase text-center',
    2: 'text-2xl sm:text-3xl mb-4 font-semibold',
    3: 'text-xl sm:text-2xl mb-3 font-semibold',
    4: 'text-base sm:text-xl mb-2 font-semibold',
    5: 'text-sm sm:text-lg font-semibold mb-1',
    6: 'text-xs sm:text-sm font-semibold',
  }

  const classes = twMerge(styles.root, sizeMap[level], className)

  return (
    <H id={id} style={style} className={classes} {...extraAttributes}>
      {children}
    </H>
  )
}

export default Heading
