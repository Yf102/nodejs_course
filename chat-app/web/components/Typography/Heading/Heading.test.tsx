import { render } from '@testing-library/react'
import Heading from './'

it('renders correctly with all props', () => {
  const { asFragment } = render(
    <Heading
      id='1234'
      level={1}
      aria-label='test heading'
      data-test='test data'
    >
      Heading level 1
    </Heading>
  )
  expect(asFragment()).toMatchSnapshot()
})

it('renders correctly without optional props', () => {
  const { asFragment } = render(<Heading level={2}>Heading level 2</Heading>)

  expect(asFragment()).toMatchSnapshot()
})
