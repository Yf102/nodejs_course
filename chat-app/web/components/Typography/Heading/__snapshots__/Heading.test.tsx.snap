// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with all props 1`] = `
<DocumentFragment>
  <h1
    aria-label="test heading"
    class="root text-3xl sm:text-4xl mb-5 font-semibold uppercase text-center"
    data-test="test data"
    id="1234"
  >
    Heading level 1
  </h1>
</DocumentFragment>
`;

exports[`renders correctly without optional props 1`] = `
<DocumentFragment>
  <h2
    class="root text-2xl sm:text-3xl mb-4 font-semibold"
  >
    Heading level 2
  </h2>
</DocumentFragment>
`;
